### DISCUSSION TITLE
How to use setQueryData to delete a list item after a mutation?

### DISCUSSION THREAD

**Discussion by ghost** [TanStack/query#2722] [Q&A] [answered]
Hell - I can't find any documentation or example for this, other than the examples that I've seen which will refetch the whole list after a delete.

Is there a way to just delete the one item out of the cache?

**Comment by TkDodo** [comment, marked as answer]
Iterate over the existing list and `filter` out the item that has been deleted

**Reply by ghost** [reply]
OK great, that helped me realize the answer - thank you. I'm just learning this library and I didn't understand that `queryClient.setQueryData` actually already lets you overwrite the entire query data when I was looking at this example: `queryClient.setQueryData(['todo', { id: variables.id }], data)`.

I thought it was only updating a single item from a list, but in reality it is setting the entire query data for a query with a key of `['todo', { id }]`. So, to delete a list item I would do `queryClient.setQueryData('todos', todos.filter(({ id }) => id !== deletedId)` as you suggested.

**Reply by michael-mcgreal** [reply]
This example and explanation was extremely helpful to me, thanks @waynebloss
